providers/logingov.go

```diff
@@ -35,7 +36,11 @@ var letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
 func randSeq(n int) string {
 	b := make([]rune, n)
 	for i := range b {
-		b[i] = letters[rand.Intn(len(letters))]
+		n, err := rand.Int(rand.Reader, big.NewInt(int64(len(letters))))
+		if err != nil {
+			panic(err)
```

Comment:
Let's return an error instead of `panic`ing here.

In the `NewLoginGovProvider` usage of this, I think it is then OK to panic if this goes bad (since this is during startup and the `New` interface for providers that consumes this doesn't do errors).

But in `Redeem` usage of this, we want to bubble up an error appropriately so the program doesn't crash but an error 500 is appropriately returned to the end user.